### PR TITLE
Fabo/timelines

### PR DESCRIPTION
**Pull request by faboweb**
Adds the democracy voting end
Adds the treasury voting end (unsure about this)

### CODE REVIEW

File: api/lib/source/polkadotV0-source.js [hunk 1]
```diff
@@ -775,6 +775,16 @@ class polkadotAPI {
       })
     )
     const votesSum = proposal.seconds.length
+
+    // mimicing the spend period in polkadot UI (not sure this is correct)
```
Comment (Bitcoinera) [inline api/lib/source/polkadotV0-source.js:779]: How can we confirm? There is no explorer displaying this information, is there?

Comment (faboweb) [reply api/lib/source/polkadotV0-source.js:779]: check proposals in motions and compare

Comment (Bitcoinera) [reply api/lib/source/polkadotV0-source.js:779]: and for democracy?

Comment (faboweb) [reply api/lib/source/polkadotV0-source.js:779]: I checked polkadot.js UI

Comment (Bitcoinera) [reply api/lib/source/polkadotV0-source.js:779]: I don't see this information anywhere in there :eyes: 

image

File: api/lib/source/polkadotV0-source.js [hunk 2]
```diff
@@ -990,6 +1012,32 @@ class polkadotAPI {
...
+          /* 6s is the average block duration for both Kusama and Polkadot */
+
+          nextSpendingBlockHeightDiff * 6 * 1000
+      )
+      console.log(
+        height,
+        spendPeriod.toNumber(),
+        nextSpendingBlockHeightDiff,
+        nextSpendingBlockTime
+      )
```
Comment (Bitcoinera) [inline api/lib/source/polkadotV0-source.js:1032]: code leftover?

Comment (Bitcoinera) [inline api/lib/source/polkadotV0-source.js:1032]:
```suggestion
```

File: api/lib/source/polkadotV0-source.js [hunk 3]
```diff
@@ -794,6 +794,18 @@ class polkadotAPI {
       })
     )
     const votesSum = proposal.seconds.length
+
+    // mimicing the spend period in polkadot UI (not sure this is correct)
+    const height = this.store.latestHeight
+    const launchPeriod = api.consts.democracy.launchPeriod // every x blocks the chain is upgraded
```
Comment (Bitcoinera) [inline api/lib/source/polkadotV0-source.js:800]: Nice!

**Comment by Bitcoinera** [comment]
It seems I had myself some concept errors in my timelines (referendum). You can see the conversation [here](https://matrix.to/#/!HzySYSaIhtyWrwiwEV:matrix.org/$1599755911180707MJvCN:matrix.org?via=matrix.parity.io&via=matrix.org&via=web3.foundation).

Will open another PR related to this

**Comment by Bitcoinera** [comment]
Maybe we need to close this PR :shrug: 

Jaco [said to me on Riot](https://matrix.to/#/!HzySYSaIhtyWrwiwEV:matrix.org/$15997563091627ZJOZv:matrix.parity.io?via=matrix.parity.io&via=matrix.org&via=web3.foundation) there is no expiry date for democracies

File: api/lib/source/polkadotV0-source.js (hunk 1 above)
Comment (faboweb) [reply api/lib/source/polkadotV0-source.js:798]: image

**Pull request closed without merging**